addon/components/ember-slider.js

```diff
@@ -24,27 +31,35 @@ export default Ember.Component.extend({
 				self.set('value', ui.value);
 			},
 			change : function(event, ui) {
-				(target.sendAction) ? target.sendAction('changeAction', ui.value) :	target.send(self.get('changeAction'), ui.value);
+				(target.sendAction) ? target.sendAction('changeAction', ui.value) : target.send(self.get('changeAction'), ui.value);
 			}
 		});
 
 		this.registerListeners();
 	},
+	
+	proxySlider: function (target, key) {
+		this.$().slider('option', key, this.get(key));
+	},
 
 	registerListeners: function () {
-		var props = ['animate', 'disabled', 'max',
-			'min', 'orientation', 'range', 'step',
-			'value', 'values'];
 		/*jshint loopfunc:false*/
-		for (var i = 0; i < props.length; i++) {
-			this.addObserver(props[i], this, function (target, key) {
-				this.$().slider('option', key, this.get(key));
-			}.bind(this));
+		for (var i = 0, len = props.length; i < len; i++) {
+			this.addObserver(props[i], this, this.proxySlider);
+		}
+		/*jshint loopfunc:true*/
+	},
+	
+	unregisterListeners: function () {
+		/*jshint loopfunc:false*/
+		for (var i = 0, len = props.length; i < len; i++) {
+			this.removeObserver(props[i], this, this.proxySlider);
 		}
 		/*jshint loopfunc:true*/
 	},
 
-	destroyEasyPie: function() {
+	destroyEasyPie: Ember.on('willDestroyElement', function() {
```

Comment:
It should have the right target



—
Sent from Mailbox

On Fri, Mar 13, 2015 at 11:01 AM, Travis Hoover <notifications@github.com>
wrote:

>>  		}
>>  		/*jshint loopfunc:true*/
>>  	},
>>  
>> -	destroyEasyPie: function() {
>> +	destroyEasyPie: Ember.on('willDestroyElement', function() {
> this should probably just go inside of the normal willDestroyElement function
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/yinaw/ember-slider/pull/9/files#r26408560